### PR TITLE
Adding AfterMigrationLoad Event

### PR DESCRIPTION
**Pull request by KhoiKy**

### CODE REVIEW

File: Rivet/Rivet.psd1 [hunk 1]
```diff
@@ -11,7 +11,7 @@
     RootModule = 'Rivet.psm1'
 
     # Version number of this module.
-    ModuleVersion = '0.13.0'
+    ModuleVersion = '0.14.0'
```
Comment (decoyjoe) [inline Rivet/Rivet.psd1:14]: Let's also move all the release notes to `CHANGELOG.md` file.
1. Rename the existing `RELEASE_NOTES.txt` file to `CHANGELOG.md`
2. Go through the all the Rivet releases [on the PowerShell Gallery](https://www.powershellgallery.com/packages/Rivet/) and add the release notes for all versions whose release notes are missing from the `CHANGELOG.md` file.
3. Update the ReleaseNotes property in this file to be a link to the `CHANGELOG.md` file on GitHub.

File: Rivet/RivetSamples/Complete-Migration.ps1 [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
+function Complete-Migration
+{
+    [CmdletBinding()]
+    [Rivet.Plugin([Rivet.Events]::AfterMigrationLoad)]
+    param(
+        [Parameter(Mandatory)]
+        # The migration the operation is part of.
```
Comment (decoyjoe) [inline Rivet/RivetSamples/Complete-Migration.ps1]: Nit: Parameter doc should be above the `[Parameter()]` attribute.

File: Rivet/RivetSamples/Complete-Migration.ps1 [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        [Rivet.Migration] $Migration
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+    
+    $problems = $false 
+
+    if( $problems )
```
Comment (decoyjoe) [inline Rivet/RivetSamples/Complete-Migration.ps1:20]: I'm not sure how this example is supposed to work. How does `$problems` get assigned?

File: Rivet/RivetSamples/Complete-Migration.ps1 [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    {
+        throw ('There were errors running ''{0}''.  Please see previous errors for details.' -f $Migration.Name)
```
Comment (decoyjoe) [inline Rivet/RivetSamples/Complete-Migration.ps1]: Nit: Our [current PowerShell style](https://github.com/webmd-health-services/Standards/tree/main/PowerShell) is to do string interpolation like this:

```powershell
throw "There were errors running ""$($Migration.Name)"". Please see previous errors for details."
```

File: Rivet/en-US/about_Rivet_Plugins.help.txt [hunk 5]
```diff
@@ -114,6 +115,36 @@ LONG DESCRIPTION
...
+    The `AfterMigrationLoad` event occurs after all operations in a migration
+    have been applied, but before the transaction has been commited. Use this
+    event for any validations that require a fully loaded migration.
+
+    These plugins get passed the `Migration` parameter which is the migration
+    object currently being loaded.
+
+    Plugins that respond to this event *must* have the `Migration` parameter.
+
+    #### Example 1
```
Comment (decoyjoe) [inline Rivet/en-US/about_Rivet_Plugins.help.txt]: Nit: There's only one example so I'd drop the `1`

**Review by decoyjoe — CHANGES_REQUESTED** [review]

File: Rivet/RivetSamples/Complete-Migration.ps1 (hunk 3 above)
Comment (KhoiKy) [reply Rivet/RivetSamples/Complete-Migration.ps1:20]: This is just a Rivet Sample for the new `AfterMigrationLoad` Rivet Event. I didn't want to put some random validation in here so I just left it bare bones. I've added a comment into this file now to show where validation should be happening.

**Review by decoyjoe — APPROVED** [review]

**Pull request merged**